DisCatSharp.ApplicationCommands/Workers/RegistrationWorker.cs

```diff
@@ -0,0 +1,400 @@
+
+        internal static bool CheckNotAnyOrNull(List<DiscordApplicationCommand> list)
+            => list == null || !list.Any();
+
+        internal static bool CheckNotAnyOrNull(List<ulong> list)
+            => list == null || !list.Any();
+
+        internal static bool CheckNotAnyOrNull(Dictionary<ulong, DiscordApplicationCommand> dict)
+            => dict == null || !dict.Any() || dict.Keys == null || !dict.Keys.Any();
+
+        internal static bool CheckAnyAndNotNull(List<DiscordApplicationCommand> list)
+            => list != null && list.Any();
+
+        internal static bool CheckAnyAndNotNull(List<ulong> list)
+            => list != null && list.Any();
+
+        internal static bool CheckAnyAndNotNull(Dictionary<ulong, DiscordApplicationCommand> dict)
+            => dict != null && dict.Any() && dict.Keys != null && dict.Keys.Any();
+
+        private static List<ulong> BuildGuildDeleteList(ulong guildId, List<DiscordApplicationCommand> updateList)
+        {
+            if (ApplicationCommandsExtension._guildDiscordCommands == null || !ApplicationCommandsExtension._guildDiscordCommands.Any())
+            {
+                return null;
+            }
+            else if (updateList == null)
+            {
+                var discord = ApplicationCommandsExtension._guildDiscordCommands.Where(l => l.Key == guildId).First().Value;
+
+                if (discord != null)
+                {
+                    List<ulong> InvalidCommandIds = new();
+                    foreach (var cmd in discord)
+                    {
+                        InvalidCommandIds.Add(cmd.Id);
+                    }
+
+                    return InvalidCommandIds;
+                }
+                else
+                {
+                    return null;
+                }
+            }
+            else
+            {
+                var discord = ApplicationCommandsExtension._guildDiscordCommands.Where(l => l.Key == guildId).First().Value;
+
+                if (discord != null)
+                {
+                    List<ulong> InvalidCommandIds = new();
+                    foreach (var cmd in discord)
+                    {
+                        if (!updateList.Any(ul => ul.Name == cmd.Name))
+                            InvalidCommandIds.Add(cmd.Id);
+                    }
+
+                    return InvalidCommandIds;
+                }
+                else
+                {
+                    return null;
+                }
+            }
+        }
+
+        private static List<DiscordApplicationCommand> BuildGuildCreateList(ulong guildId, List<DiscordApplicationCommand> updateList)
+        {
+            if (ApplicationCommandsExtension._guildDiscordCommands == null || !ApplicationCommandsExtension._guildDiscordCommands.Any())
+            {
+                return updateList;
+            }
+            else if (updateList == null)
+            {
+                return null;
+            }
+            else
```

Comment:
remove the else, indent code inside to left. It's assumed that reaching this point means the above hasn't been met